test_compare/cmp.py

```diff
@@ -0,0 +1,679 @@
+            for name in fails:
+                print(f"{styles[level]}{prefixes[level]}{name}{ESCAPE_RESET}")
+        return
+    for name, element in fails.items():
+        print(f"{styles[level]}{prefixes[level]}{name}{ESCAPE_RESET}")
+        if args.verbose > level:
+            print_fails(element, args, level + 1, parent)
+            if level == 0:
+                print()
+
+
+def generate_status_rows(statuses, args, level=0):
+    styles = [ESCAPE_BOLD + ESCAPE_ITALIC, ESCAPE_BOLD, ESCAPE_BOLD, ""]
+    separators = ["║", "║", "┃", "┆"]
+    prefixes = ["Target: ", "-", " -", "  -"]
+    rows = []
+    for node in statuses:
+        if args.verbose == level and node["status_old"] == node["status_new"]:
+            continue
+        current_row = StatusRow(
+            styles[level],
+            separators[level],
+            f"{prefixes[level]}{node['name']}",
+            node["status_old"],
+            node["status_new"],
+        )
+        children_rows = None
+        if args.verbose > level:
+            children_rows = generate_status_rows(node["children"], args, level + 1)
+        if node["status_old"] != node["status_new"] or children_rows:
+            if level == 0 and (args.verbose > 0 or not rows):
+                if rows:
+                    rows.append(StatusRow())
+                rows.append(StatusRow("", "", "H"))
+            rows.append(current_row)
+            if args.verbose > level:
+                rows.extend(children_rows)
+    return rows
+
+
+def generate_time_rows(times, args, level=0):
+    styles = [ESCAPE_BOLD + ESCAPE_ITALIC, ESCAPE_BOLD, ESCAPE_BOLD, ""]
+    separators = ["║", "║", "┃", "┆"]
+    prefixes = ["Target: ", "-", " -", "  -"]
+    rows = []
+    for node in times:
+        go_deeper = args.verbose > level and (level != 2 or not node["single_case"])
+        if not go_deeper and not filter_display(node, level, args):
+            continue
+        current_row = make_row(node, styles[level], separators[level], prefixes[level], args)
+        children_rows = None
+        if go_deeper:
+            children_rows = generate_time_rows(node["children"], args, level + 1)
+        if not go_deeper or children_rows:
+            if level == 0 and (args.verbose > 0 or not rows):
+                if rows:
+                    rows.append(Row())
+                rows.append(Row("", "", "H"))
+            rows.append(current_row)
+            if go_deeper:
+                rows.extend(children_rows)
+    return rows
+
+
+def main():
+    args: Args = process_args(parse_args())
+    for file, name in [(args.file_old, "old"), (args.file_new, "new")]:
+        fails = find_fails(file, args, unfiltered=True)
+        if fails:
+            fails_filtered = find_fails(file, args)
+            if args.show_fails:
+                print(f"Failed tests in {name} file:")
+                print_fails(fails_filtered, args)
+            else:
+                print(
```

Comment:
this is a script for diff'ing 2 JUnit XML files - use-case wise - I would like to see that the test started (or stopped) failing without any extra params as this is crucial info.